Observer/Predispatch.php

```diff
@@ -59,7 +59,11 @@ public function execute(Observer $observer)
             }
 
             // Pass on any query parameters to the configurable product's URL.
-            $query = $request->getQuery() ? '?' . http_build_query($request->getQuery()) : '';
+            $query = $request->getQuery();
+            if (is_object($query)) {
+                $query = $query->toArray();
```

Comment:
Is the `toArray()` method always available? What happens if we'd just `$query = (array) $query;`?

Reply:
Not sure, you could perhaps add a check to see if it's an object, but you can't cast to array.